### PR TITLE
Sentry integration

### PR DESCRIPTION
**Pull request by jojolepro**
Adds Sentry integration

- [X] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: .sentry_dsn.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+https://6bc2b763b60e4651ac237871056eb21b@sentry.io/1445091
```
Comment (LucioFranco) [inline .sentry_dsn.txt]: probably dont want this?

File: build.rs [hunk 2]
```diff
@@ -1,8 +1,90 @@
...
+            true
+        },
+        false => {
+            let _ = file.write_all(b"false");
+            false
+        }
+    }
+}
+
+fn load_sentry_dsn() {
```
Comment (LucioFranco) [inline build.rs]: should just load it from env var?

File: src/app.rs [hunk 3]
```diff
@@ -477,6 +486,10 @@ where
         info!("Version: {}", env!("CARGO_PKG_VERSION"));
         info!("Platform: {}", env!("VERGEN_TARGET_TRIPLE"));
         info!("Amethyst git commit: {}", env!("VERGEN_SHA"));
+        if let Some(sentry) = option_env!("SENTRY_DSN") {
+            info!("Sentry DSN: {}", sentry);
```
Comment (LucioFranco) [inline src/app.rs]: and probably dont want to print this atleast not at an info level?

File: .sentry_dsn.txt (hunk 1 above)
Comment (fhaynes) [reply .sentry_dsn.txt]: We do. It is the only way to send events.

File: build.rs (hunk 2 above)
Comment (fhaynes) [reply build.rs]: No, we want people to be able to use it without having to worry about finding the DSN

File: src/app.rs (hunk 3 above)
Comment (fhaynes) [reply src/app.rs]: No, it is fine for the DSN to be public. The service has rate limiting and IP restriction built in if we need it.

**Comment by fhaynes** [comment]
@jojolepro cargo fmt and good to go

**Comment by jojolepro** [comment]
bors retry

**Comment by torkleyy** [comment]
How will this work for game deployment? Also, what happens for non-interactive builds?

**Comment by fhaynes** [comment]
```
How will this work for game deployment?
```
For now, it shouldn't be deployed into a release version of a game to customers. It is meant as a debugging tool for our direct users, not their customers.

```
Also, what happens for non-interactive builds?
```
It should check the files as normal, and default to not including it if the config files aren't present.

**Pull request merged**